core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1122,30 +1119,29 @@ public File download(DownloadJob job, URL src) throws IOException {
                 con.setReadTimeout(PLUGIN_DOWNLOAD_READ_TIMEOUT);
                 
                 int total = con.getContentLength();
-                in = new CountingInputStream(con.getInputStream());
                 byte[] buf = new byte[8192];
                 int len;
 
                 File dst = job.getDestination();
                 File tmp = new File(dst.getPath()+".tmp");
-                out = Files.newOutputStream(tmp.toPath());
-                if (sha1 != null) {
-                    out = new DigestOutputStream(out, sha1);
-                }
 
                 LOGGER.info("Downloading "+job.getName());
```

Comment:
Maybe this log message should be moved as well. otherwise the logic slightly changes here. Seems to be NIT